src/dmd/frontend.h

```diff
@@ -750,7 +750,7 @@ struct Loc
     const char* filename;
     uint32_t linnum;
     uint32_t charnum;
-    const char* toChars(bool showColumns, uint8_t messageStyle) const;
+    const char* toChars(bool showColumns = global.params.showColumns, uint8_t messageStyle = global.params.messageStyle) const;
```

Comment:
This will trigger a forward reference error, since `global` (the variable) is declared at L7118

Reply:
Done, the added code will now properly emit forward references (but there are some more issues in the existing code which i would rather defer into another PR)